### ISSUE TITLE
Find out if patron PIN auth is unblocked with the latest FOLIO update

### ISSUE THREAD

**Issue by cbeer** [sul-dlss/mylibrary#788]
We're supposedly using `mod-users 19.0` with support for doing patron PIN auth. We should verify these APIs work as expected, and unblock:

https://github.com/sul-dlss/mylibrary/issues/739

**Comment by thatbudakguy** [comment]
It appears that the routes to set and delete patron PINs work as expected, but PINs cannot actually be validated, as the `/patron-pin/verify` route is an internal-only API. It returns:
> No suitable module found for path /patron-pin/verify for tenant sul

It's possible that the only way to verify a PIN is to actually execute a login on behalf of the user using the PIN as the password, and then (on a success) throw away the login token and continue to execute actions as the `mylibrary` user.

Notably, this [Jira page on the issue](https://wiki.folio.org/pages/viewpage.action?pageId=65111663) mentions (at the very bottom):
> The functionality for an email based PIN reset will be addressed later and should work similar to FOLIOs password reset mechanism.

**Comment by thatbudakguy** [comment]
Some more things I learned after pairing with @shelleydoljack and @jgreben:

- The PIN functionality in FOLIO is unfinished. In order to support people logging in using something other than SSO, we need to have their user be assigned a password in FOLIO. Thus, current courtesy patrons will use their barcode as a username, and their PIN as a password.
- We confirmed that all of the courtesy patrons migrated to folio-test already had their username set as their barcode. However, their PINs were not migrated to passwords. **We need a migration strategy to address this.**
- We should probably be using the [`/bl-users` endpoint](https://s3.amazonaws.com/foliodocs/api/mod-users-bl/r/mod-users-bl.html#bl_users_settings_myprofile_password_post) in mylibrary. It describes itself as a "front end for mod-users and mod-permissions" and seems capable of calling out to many useful things under the hood. It also has useful functions for resetting a user's password.
- We might want to change the language around "PIN"s in mylibrary, since they aren't really a thing in FOLIO (or alternatively we could keep the "PIN" language, with the knowledge that "PIN" just means "non-SSO password").
